### PR TITLE
Improve with firebase cloud messaging

### LINKED ISSUE #20826: Custom server is not necessary in with-firebase-cloud-messaging
**What example does this report relate to?**

with-firebase-cloud-messaging

**What version of Next.js are you using?**

10.0.4

**What version of Node.js are you using?**

12.18.2

**What browser are you using?**

Chrome and Firefox

**What operating system are you using?**

Ubuntu 18

**How are you deploying your application?**

Vercel

**Describe the Bug**

In the with-firebase-cloud-messaging example, was created a custom server to serve some static files from /.next/ folder in the root scope. But I really dont understand the necessity of this custom server and it disable important performance optimizations (according to this doc page https://nextjs.org/docs/advanced-features/custom-server).

So i created a POC to test if the FCM works without the custom server and works fine (i tested in localhost and on Vercel).

**Expected Behavior**

The example should not have the custom server.

**To Reproduce**

I created this repository without the custom server: https://github.com/VictorAssis/with-firebase-cloud-messaging

My question is: does this custom server have any functionality that I am not noticing?

If that is no longer needed, I can create a fork to update the example.

### PR DESCRIPTION
**Pull request by VictorAssis**
Fix #20826.

Remove unnecessary dependencies and custom server to implement Firebase Cloud Messaging.

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| buildDuration | 10.9s | 10.6s | -229ms |
| nodeModulesSize | 80.8 MB | 80.8 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.111 | 2.055 | -0.06 |
| / avg req/sec | 1184.23 | 1216.56 | +32.33 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.415 | 1.367 | -0.05 |
| /error-in-render avg req/sec | 1766.23 | 1828.36 | +62.13 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| 677f882d2ed8..89aa.js gzip | 13.1 kB | 13.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-dba4ec1..27c0.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.4 kB | 59.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| polyfills-81..14d7.js gzip | 31.2 kB | 31.2 kB | ✓ |
| Overall change | 31.2 kB | 31.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| _app-b6fc6bc..222c.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-e2ffa..0f3f.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-010c20..8411.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-d979103..c5ff.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| index.html gzip | 614 B | 614 B | ✓ |
| link.html gzip | 619 B | 619 B | ✓ |
| withRouter.html gzip | 606 B | 606 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| buildDuration | 12.8s | 12.4s | -381ms |
| nodeModulesSize | 80.8 MB | 80.8 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| 677f882d2ed8..89aa.js gzip | 13.1 kB | 13.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-dba4ec1..27c0.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.4 kB | 59.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| polyfills-81..14d7.js gzip | 31.2 kB | 31.2 kB | ✓ |
| Overall change | 31.2 kB | 31.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| _app-b6fc6bc..222c.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-e2ffa..0f3f.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-010c20..8411.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-d979103..c5ff.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.18 MB | 5.18 MB | ✓ |

</details>

</details>
Commit: 33dabe0355c7ab29916f524e748e99a7b8394163

File: examples/with-firebase-cloud-messaging/package.json [hunk 1]
```diff
@@ -2,21 +2,18 @@
...
   "scripts": {
-    "dev": "node server.js",
+    "dev": "next dev",
     "build": "next build",
-    "start": "cross-env NODE_ENV=production node server.js"
+    "start": "next start"
   },
   "dependencies": {
-    "cross-env": "^7.0.2",
     "express": "^4.14.0",
```
Comment (lachlanjc) [inline examples/with-firebase-cloud-messaging/package.json:10]: Can probably drop Express too?

Comment (VictorAssis) [reply examples/with-firebase-cloud-messaging/package.json]: Yes! I forgot the express. Updated.

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| buildDuration | 9.3s | 9.5s | ⚠️ +195ms |
| nodeModulesSize | 80.8 MB | 80.8 MB | ✓ |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 1.837 | 1.972 | ⚠️ +0.14 |
| / avg req/sec | 1361.1 | 1267.94 | ⚠️ -93.16 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.198 | 1.213 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 2086.76 | 2060.31 | ⚠️ -26.45 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| 677f882d2ed8..89aa.js gzip | 13.1 kB | 13.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-dba4ec1..27c0.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.4 kB | 59.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| polyfills-81..14d7.js gzip | 31.2 kB | 31.2 kB | ✓ |
| Overall change | 31.2 kB | 31.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| _app-b6fc6bc..222c.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-e2ffa..0f3f.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-010c20..8411.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-d979103..c5ff.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| index.html gzip | 614 B | 614 B | ✓ |
| link.html gzip | 619 B | 619 B | ✓ |
| withRouter.html gzip | 606 B | 606 B | ✓ |
| Overall change | 1.84 kB | 1.84 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| buildDuration | 11.4s | 11.4s | ⚠️ +5ms |
| nodeModulesSize | 80.8 MB | 80.8 MB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| 677f882d2ed8..89aa.js gzip | 13.1 kB | 13.1 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-dba4ec1..27c0.js gzip | 6.63 kB | 6.63 kB | ✓ |
| webpack-50be..df5b.js gzip | 751 B | 751 B | ✓ |
| Overall change | 59.4 kB | 59.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| polyfills-81..14d7.js gzip | 31.2 kB | 31.2 kB | ✓ |
| Overall change | 31.2 kB | 31.2 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| _app-b6fc6bc..222c.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-e2ffa..0f3f.js gzip | 3.46 kB | 3.46 kB | ✓ |
| hooks-010c20..8411.js gzip | 887 B | 887 B | ✓ |
| index-bbee2f..528b.js gzip | 227 B | 227 B | ✓ |
| link-d979103..c5ff.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..bf84.js gzip | 303 B | 303 B | ✓ |
| withRouter-a..5826.js gzip | 302 B | 302 B | ✓ |
| Overall change | 8.09 kB | 8.09 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| _buildManifest.js gzip | 323 B | 323 B | ✓ |
| Overall change | 323 B | 323 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | VictorAssis/next.js improve-with-firebase-cloud-messaging | Change |
| - | - | - | - |
| _error.js | 1 MB | 1 MB | ✓ |
| 404.html | 2.67 kB | 2.67 kB | ✓ |
| hooks.html | 1.92 kB | 1.92 kB | ✓ |
| index.js | 1 MB | 1 MB | ✓ |
| link.js | 1.06 MB | 1.06 MB | ✓ |
| routerDirect.js | 1.05 MB | 1.05 MB | ✓ |
| withRouter.js | 1.05 MB | 1.05 MB | ✓ |
| Overall change | 5.18 MB | 5.18 MB | ✓ |

</details>

</details>
Commit: e164b1bf48673fbb2452a193b34eca13ed96d561

**Comment by timneutkens** [comment]
Thanks!

**Pull request merged**